main/integration-tests/__test__/api-tests/workspace-service-catalogs/create-workspace-service-catalog.test.js

```diff
@@ -166,4 +179,90 @@ describe('Create workspace-service-catalog scenarios', () => {
+      // Things to do
+      // 1. Create user
+      const admin1Session = await setup.createAdminSession();
+      const username = setup.gen.username();
+      const defaultUser = admin1Session.resources.users.defaults({ username });
+      await expect(admin1Session.resources.users.create(defaultUser)).resolves.toMatchObject({
+        username,
+      });
+      // 2. Assign Project ID to user (assigned by default)
+
+      // 3. Create My Study, Org Study, Data Source study
+      const studyIds = [];
+      let studyId = setup.gen.string({ prefix: `create-study-ray-my-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'My Studies' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      studyId = setup.gen.string({ prefix: `create-study-ray-org-study` });
+      await expect(
+        admin1Session.resources.studies.create({ id: studyId, name: studyId, category: 'Organization' }),
+      ).resolves.toEqual(
+        expect.objectContaining({
+          id: studyId,
+        }),
+      );
+      studyIds.push(studyId);
+
+      // 4. Create workspace with the above studies
+      const workspaceName = setup.gen.string({ prefix: 'workspace-sc-test' });
+
+      const env = await admin1Session.resources.workspaceServiceCatalogs.create({
+        name: workspaceName,
+        envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+        envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+        studyIds,
+        description: 'assignment',
+        projectId: setup.defaults.project.id,
+        cidr: '54.240.196.186/32',
+      });
+      expect(env).toEqual(
+        expect.objectContaining({
+          name: workspaceName,
+          envTypeId: setup.defaults.envTypes.ec2Linux.envTypeId,
+          envTypeConfigId: setup.defaults.envTypes.ec2Linux.envTypeConfigId,
+          studyIds,
+        }),
+      );
+
+      const workflows = await admin1Session.resources.workflows
+        .versions('wf-provision-environment-sc')
+        .version(1)
+        .instances()
+        .get();
+      let result;
+      for (const w of workflows) {
+        if (
+          w.input.envId === env.id &&
+          w.input.envTypeId === env.envTypeId &&
+          w.input.envTypeConfigId === env.envTypeConfigId
+        ) {
+          result = w;
+        }
+      }
+      while (result.wfStatus !== 'done' && result.wfStatus !== 'error') {
+        // look at retry logic
+        await new Promise(r => setTimeout(r, 1000 * 30));
+        const getPromises = await Promise.all([
+          admin1Session.resources.workflows
+            .versions('wf-provision-environment-sc')
+            .version(1)
+            .instances()
+            .instance(result.id)
+            .get(),
+        ]);
+        result = getPromises[0];
```

Comment:
Are you missing an expect statement here? These lines don't seem to be doing anything for the test.

Reply:
I initially had the array access operator right after the async call. This led to issues, so I just assigned it in a different statement, line 264.

Reply:
In reason why you want to do Promise.all?  I think you can write a simple while loop with await delay and await admin1Session.resource.workflows...get()

Reply:
I am also thinking that you can move this to a helper method

Reply:
also in response to the first comment, i saw the assignment but there's nothing that checks the actual value of the result, so it's not actually being tested.

Reply:
I created utility functions for polling and moved it to `workflow-version.js`, and modified triggerAndWait to reduce redundancy.